lib/basedriver/capabilities.js

```diff
@@ -0,0 +1,111 @@
+import _ from 'lodash';
+import { validator } from './desired-caps';
+import { util } from 'appium-support';
+
+// Takes primary caps object and merges it into a secondary caps object.
+// (see https://www.w3.org/TR/webdriver/#dfn-merging-capabilities)
+function mergeCaps (primary = {}, secondary = {}) {
+  let result = Object.assign(primary);
```

Comment:
This needs to be `let result = Object.assign({}, primary);` or else it is not doing anything:
```
appium-xcuitest-driver → git master* → node                                                  4.8.2
> a = {s: 't'}
{ s: 't' }
> b = Object.assign(a)
{ s: 't' }
> b
{ s: 't' }
> a
{ s: 't' }
> b.f = 'g'
'g'
> a
{ s: 't', f: 'g' }
> b
{ s: 't', f: 'g' }
> c = Object.assign({}, a)
{ s: 't', f: 'g' }
> c.a = 'b'
'b'
> c
{ s: 't', f: 'g', a: 'b' }
> a
{ s: 't', f: 'g' }
> b
{ s: 't', f: 'g' }
>
```